ДЗ 21-го спринта

powerrewop opened the pull request:
ДЗ 21-го спринта, 19-я когорта Андроид разработки
исправления замечаний:

Баг. Окружение: эмулятор Pixel 6, 33 Api
Кейс: перейти на экран "Поиск" -> выполнить поиск -> нажать на любой найденный трек -> открылся экран "Аудиоплеер" -> добавить трек в избранное и нажать кнопку "Назад" -> произошёл возврат на экран "Поиск" -> снова открыть тот же трек
ОР: кнопка «Нравится» находится в выделенном состоянии
ФР: кнопка «Нравится» находится в не выделенном состоянии.

исправлено (и для формы избранного тоже)

powerrewop committed:
ДЗ 21-го спринта

IliaAnK commented on app/src/main/java/com/practicum/playlistmaker/data/db/dao/FavTrackDao.kt:
```diff
@@ -0,0 +1,26 @@
+     fun deleteMovieEntity(trackEntity: FavTrackEntity)
+
+    @Query("SELECT * FROM fav_tracks_table ORDER BY dateFav DESC")
+     fun getFavTracks(): List<FavTrackEntity>
```
Рекомендация:
`Room` может возвращать `Flow`
Получится:
`fun getFavTracks(): Flow<List<FavTrackEntity>>`

Для `getAllIdFavTracks` **аналогично**

IliaAnK commented on app/src/main/java/com/practicum/playlistmaker/presentation/models/FavLibModel.kt:
```diff
@@ -0,0 +1,15 @@
+
+
+sealed class FavLibModel {
+    class EmptyContent : FavLibModel()
```
Рекомендация:
`class` -> `data object`

Для `LoadContent` **аналогично**

powerrewop committed:
ДЗ 21-го спринта, исправления замечаний:

Баг. Окружение: эмулятор Pixel 6, 33 Api
Кейс: перейти на экран "Поиск" -> выполнить поиск -> нажать на любой найденный трек -> открылся экран "Аудиоплеер" -> добавить трек в избранное и нажать кнопку "Назад" -> произошёл возврат на экран "Поиск" -> снова открыть тот же трек
ОР: кнопка «Нравится» находится в выделенном состоянии
ФР: кнопка «Нравится» находится в не выделенном состоянии.

исправлено (и для формы избраного тоже)
